lib/user.js

```diff
@@ -55,17 +71,41 @@ const instanceMethods = {
         });
     },
 
-    get functions() {
+    remoteMongoClient(serviceName) {
         const user = this;
-        return new Proxy({}, {
-            get(target, name) {
-                return (...args) => {
-                    return user.callFunction(name, args);
+        return {
+            db(dbName) {
```

Comment:
Is this calling a function or returning an object?

Reply:
`remoteMongoClient()` is returning an object with a single method `db()`, which returns and object with the single method `collection()` which returns a RemoteMongoDBCollection object.

It may be clearer if you see a usage example: https://github.com/realm/realm-js/blob/d13d14eb38c7b27ad7b3c5fd2ceb6d479d2f704e/tests/js/user-tests.js#L158-L165

Reply:
got it